front/src/modules/simulationResult/components/SpaceTimeChart/useGetProjectedTrainOperationalPoints.ts

```diff
@@ -40,13 +59,44 @@ const useGetProjectedTrainOperationalPoints = (
         };
         const { operational_points } = await postPathProperties(pathPropertiesParams).unwrap();
 
-        setOperationalPoints(
-          operational_points as NonNullable<PathProperties['operational_points']>
-        );
+        let operationalPointsWithAllWaypoints = operational_points as NonNullable<
+          PathProperties['operational_points']
+        >;
+        // Check if there are vias added by map click and insert them in the operational points
+        if (trainScheduleUsedForProjection.path.some((step) => 'track' in step)) {
+          trainScheduleUsedForProjection.path.forEach((step, i) => {
+            if ('track' in step) {
+              const positionOnPath = pathfindingResult.path_item_positions[i];
+              const indexToInsert = operationalPointsWithAllWaypoints.findIndex(
+                (op) => op.position >= positionOnPath
+              );
+
+              const formattedStep: NonNullable<PathProperties['operational_points']>[number] = {
+                id: step.id,
+                extensions: {
+                  identifier: {
+                    name: t('requestedPoint', { count: indexToInsert }),
+                    uic: 0,
```

Comment:
Hm. This sounds like a bit of an abuse of the `PathProperties.operational_points` type...

Would we be able to perform this conversion a bit later so that we're dealing with `PathStep` instead of this type? That way we wouldn't have to fake an OP object for something which isn't an OP?

Reply:
I don't like that either. But `later` would be inside `Manchette` component in osrd-ui, we give that array directly to the component after performing this hook.

Reply:
Oh my. So, the bad news is that I now remember that ui-manchette just duplicates the editoast types as-is… And the good news is that the result of this function is only consumed by ui-manchette and nothing else (so no need to refactor a whole lot of OSRD files when we decide to change the return type), and that we've planned to refactor ui-manchette to stop doing this: https://github.com/osrd-project/osrd-confidential/issues/637

Let's leave it at that for now :)